content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,123 @@
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To create Prisma Client extensions, use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods), and the appropriate extension type (`model`, `client`, `result` or `query`).
+
+For example, the following code uses the `model` extension type to extend the `user` model:
+
+```ts
+const xprisma = prisma.$extends({
+  name: `signUp`,  // (Optional) This extension name appears in error logs and tracing
+  model: {  // This is a model type extension
+    user: { ... } // The extension logic goes inside the curly braces
+ },
+})
+```
+
+`name` is an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+You can use [multiple independent extensions at the same time](#multiple-extensions). Each of these instances runs in is own isolated instance of Prisma Client, unless you deliberately combine or stack them.
+
+You can also [share your Prisma Client extensions](/concepts/components/prisma-client/client-extensions/shared-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+## Prerequisites
+
+To use Prisma Client extensions, you must enable the `clientExtensions` preview feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Multiple extensions
+
+Each time that you call `$extends`, you create an _extended client_. This extended client is a forked variant of the standard Prisma Client. The original client does not change.
+
+Every extended client operates independently, in an isolated instance. It cannot conflict with other extended clients or the original client.
+
+You can also combine two or more extensions into one extension, if you want the functionality from both extensions to apply in the same client instance. Note that it is possible for [combined extensions to conflict](#conflicts-in-combined-extensions).
+
+### Examples of isolated client instances
+
+Example uses of isolated client instances include the following:
+
+- Implement row-level security (RLS), where each http request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
+- Add a `user.current()` method for the `user` model to get the current logged-in user
+- Enable more verbose logging for requests if a debug cookie is set
+- Attach a unique request id to all logs so you can correlate them later
+- Remove a `delete` method from models unless the application calls the admin endpoint and the user has the necessary priviliges
```

Comment:
This feels out of place under "Multiple extensios" headline. Probably should be moved up together with the sentences above? https://github.com/prisma/docs/pull/4004/files#r1015818102

Reply:
Yes, I've moved it into the intro section, and put some more headings in to provide signposting for the now more dense intro